### PR TITLE
#12081 Fix empty string is accepted by API

### PR DESCRIPTION
**Pull request by DanieleAurilio**
### What does it do?

Check for not empty string or not null value accepted by API.

### Why is it needed?

 When I send a POST request to API with an email field type, for example: 
`{
  "data": {
    "email": ""
  }
}` 

It creates an empty object and get a 200 OK response.

### How to test it?

1. Create a collection type with a required email field
2. Send a post request with an empty string for the field
3. Now get a 400 Validation Error instead 200.

### Related issue(s)/PR(s)

Related Issue: [12081](https://github.com/strapi/strapi/issues/12081)

### CODE REVIEW

**Comment by abdonrd** [comment]
Seems related to https://github.com/strapi/strapi/issues/11882

File: packages/core/strapi/lib/services/entity-validator/validators.js [hunk 1]
```diff
@@ -166,7 +166,9 @@ const stringValidator = composeValidators(
   addUniqueValidator
 );
 
-const emailValidator = composeValidators(stringValidator, validator => validator.email());
+const emailValidator = composeValidators(stringValidator, validator =>
```
Comment (derrickmehaffy) [inline packages/core/strapi/lib/services/entity-validator/validators.js:169]: Am I missing something or does this not change anything but move something to a new line?

**Review by derrickmehaffy — CHANGES_REQUESTED** [review]

File: packages/core/strapi/lib/services/entity-validator/validators.js (hunk 1 above)
Comment (DanieleAurilio) [reply packages/core/strapi/lib/services/entity-validator/validators.js:169]: Hi @derrickmehaffy ,
fix `validator.email()` to `validator.email().required()` to line 170.

Comment (derrickmehaffy) [reply packages/core/strapi/lib/services/entity-validator/validators.js:169]: I am blind, thanks :laughing:

**Review by alexandrebodin — CHANGES_REQUESTED** [review]
Thank you for contributing, Could you add some testing for this change?

**Comment by DanieleAurilio** [comment]
>Thank you for contributing, Could you add some testing for this change?

Hi @alexandrebodin ,
i added e2e test on validator. 
Thank you

File: packages/core/strapi/lib/services/entity-validator/__tests__/email-validators.test.js [hunk 2]
```diff
@@ -37,5 +37,18 @@ describe('Email validator', () => {
...
+      const validator = strapiUtils.validateYupSchema(
+        validators.email(
+          {
+            attr: { type: 'string', required: true },
+          },
+          { isDraft: false }
+        )
+      );
+
+      expect(await validator('valid@email.com')).not.toBe('');
```
Comment (alexandrebodin) [inline packages/core/strapi/lib/services/entity-validator/__tests__/email-validators.test.js:51]: I'm not sure I understand the test here. Shouldn't you check that sending an empty string throws?

Comment (DanieleAurilio) [reply packages/core/strapi/lib/services/entity-validator/__tests__/email-validators.test.js]: Hi @alexandrebodin,
sorry first time with tests. I pushed a fix, can you review it?

**Review by alexandrebodin — APPROVED** [review]
Thank you for this PR 💯. I made very small changes to approve it :)

**Pull request merged**